### PR TITLE
to #7005: transform params to SPRING_DATASOURCE_XXX

### PR DESCRIPTION
**Pull request by zwZjut**
## Purpose of the pull request

https://github.com/apache/dolphinscheduler/issues/7005

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by zwZjut** [comment]
@kezhenxu94  please take a look at this

File: docker/build/startup-init-conf.sh [hunk 1]
```diff
@@ -26,10 +26,19 @@ echo "init env variables"
...
+export DATABASE_PORT=${DATABASE_PORT:-"5432"}
+export DATABASE_USERNAME=${DATABASE_USERNAME:-"root"}
+export DATABASE_PASSWORD=${DATABASE_PASSWORD:-"root"}
+export DATABASE_DATABASE=${DATABASE_DATABASE:-"dolphinscheduler"}
+export DATABASE_PARAMS=${DATABASE_PARAMS:-"characterEncoding=utf8"}
+# transform params to SPRING_DATASOURCE_XXX
+export SPRING_DATASOURCE_URL="jdbc:${DATABASE_TYPE}://${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_DATABASE}?${DATABASE_PARAMS}"
+export SPRING_DATASOURCE_USERNAME=${DATABASE_USERNAME}
+export SPRING_DATASOURCE_PASSWORD=${DATABASE_PASSWORD}
+export SPRING_DATASOURCE_DRIVER_CLASS_NAME=${DATABASE_DRIVER}
```
Comment (kezhenxu94) [inline docker/build/startup-init-conf.sh]: This is not needed, when the database type is set, the corresponding driver will be automatically picked up so let’s not expose the config to reduce users extra work.

**Comment by kezhenxu94** [comment]
Please also adjust the env vars in e2e test. https://github.com/apache/dolphinscheduler/blob/ef0cb466619621556d9f2e886d55707f0b0b8a8e/dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/resources/docker/tenant/docker-compose.yaml#L24-L25

**Comment by zwZjut** [comment]
> Please also adjust the env vars in e2e test.
> 
> https://github.com/apache/dolphinscheduler/blob/ef0cb466619621556d9f2e886d55707f0b0b8a8e/dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/resources/docker/tenant/docker-compose.yaml#L24-L25

the number h2 url  of patterns  is too many, just skip it and use application-h2.yaml default config.

**Comment by kezhenxu94** [comment]
Ok. LGTM @CalvinKirs

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**